### PR TITLE
Error handling in downloadThread

### PR DESCRIPTION
**Pull request by Sujanadh**
# Update
fixed issue prevailing in handling the json returned from listSubmission.

### CODE REVIEW

File: osm_fieldwork/OdkCentral.py [hunk 1]
```diff
@@ -64,12 +64,12 @@ def downloadThread(project_id: int, xforms: list, odk_credentials: dict, filters
         form = OdkForm(odk_credentials["url"], odk_credentials["user"], odk_credentials["passwd"])
         # submissions = form.getSubmissions(project_id, task, 0, False, True)
         subs = form.listSubmissions(project_id, task, filters)
-        if type(subs) == dict:
+        if subs == {}:
```
Comment (spwoodcock) [inline osm_fieldwork/OdkCentral.py:67]: `if subs == {}`, then `subs["value"]` will not exist, so the log.error will fail

Comment (spwoodcock) [reply osm_fieldwork/OdkCentral.py:67]: What format is the error JSON if getting the submission fails?

You could probably do something like:
```python
if isinstance(subs, dict) and (error_code := subs.get("code", False)):
    log.error(
        f"Failed to get submissions for project ({project_id}) task ({task})"
        f" with error code {error_code}: {subs.get('message')}"
    )
    continue
if len(subs.get("value"), []) > 0:
    data += subs["value"]
```

**Review by spwoodcock — CHANGES_REQUESTED** [review]
see comment

File: osm_fieldwork/OdkCentral.py (hunk 1 above)
Comment (Sujanadh) [reply osm_fieldwork/OdkCentral.py:67]: yes you are right.
our listSubmissions() looks like this:
```python
url = f"{self.base}projects/{projectId}/forms/{xform}.svc/Submissions"
result = self.session.get(url, params=filters, verify=self.verify)
if result.ok:
    self.submissions = result.json()
    return self.submissions

return {}
```
if result is not ok that means status code returned by the server is not in the range of 200-299,
empty dict is returned. so there wont be message and status code.
is this log okay? 
```python
if subs == {}:
    log.error(f"Failed to get submissions for project ({project_id}) task ({task})")
    continue
```

**Pull request merged**